### DISCUSSION TITLE
Weekly #7

### DISCUSSION THREAD

**Discussion by Ninetilt** [Ninetilt/Project-TDC#15] [General]
# **Übersicht**
Dieses Weekly hat wieder einmal das Ende des letzten Sprints und den Beginn des nächsten Weeklys eingerufen. Dazu haben wir auch diesmal wieder ein Paar Dinge zu sagen. Davor haben wir aber einige organisatorische Update für euch. Wie es diese Woche die Aufgabe haben wir uns Gedanken zu den architektonischen Ansätzen unseres Projekts Gedanken gemacht. Zusätzlich haben wir auch Fortschritte zum Design unseres Projekts, die wir euch hier näher bringen wollen.

## **Architektur**
Wie es diese Woche die Aufgabe war haben wir uns fünf Qualitätsszenarien zu Projekt-TDC ausgedacht. Diese lauten wie folgt:
- Zuverlässigkeit: Stabilität der App ([S01](https://github.com/Ninetilt/Project-TDC/blob/main/docs/quality/S01.pdf))
- Leistungsfähigkeit: Schnelles Erstellen einer Liste ([S02](https://github.com/Ninetilt/Project-TDC/blob/main/docs/quality/S02.pdf))
- Benutzerfreundlichkeit: und Login ([S03](https://github.com/Ninetilt/Project-TDC/blob/main/docs/quality/S03.pdf))
- Sicherheit: Passwort zurücksetzen ([S04](https://github.com/Ninetilt/Project-TDC/blob/main/docs/quality/S04.pdf))
- Interoperabilität: Freundesliste und Freundesverwaltung ([S05](https://github.com/Ninetilt/Project-TDC/blob/main/docs/quality/S05.pdf))

Für mehr Informationen führen die Links zu detaillierten Beschreibungen zu jedem einzelnen Szenario.

Hier seht ihr die Qualitätsszenarien auch noch einmal als [Qualitätsbaum ](https://github.com/Ninetilt/Project-TDC/blob/main/docs/quality/Qualitaetsbaum.pdf)aufgeführt:

![Qualitaetsbaum](https://github.com/user-attachments/assets/98f5f23f-1dee-4d22-a761-e0cfb15fcf72)

Der hauptächlische Architekturstil, den wir für unsere App gewählt haben ist die "[serviceorientierte Archritektur](https://github.com/Ninetilt/Project-TDC/blob/main/docs/architecture/ADR1.pdf)". Diese wird den Hauptteil unserer App ausmachen. Zusätzlich dazu wollen wir mit [RESTful API](https://github.com/Ninetilt/Project-TDC/blob/main/docs/architecture/ADR3.pdf) und [asynchronen API-Aufrufen](https://github.com/Ninetilt/Project-TDC/blob/main/docs/architecture/ADR2.pdf) arbeiten. Mehr Informationen und unsere Gedankengänge hinter der Wahl könnt ihr über die Links genauer erforschen.

## **Design**
Diesen Sprint hat sich unser Designer die Arbeit gemacht einige Aspekte der Corporate Identity zu definieren. Nach Besprechung mit dem Team sind wir auf folgendes Ergebnis gekommen:
- TDC wird ein Maskottchen als Markenzeichen besitzen
- Das Maskottchen ist ein stilisiertes Kapuziner-Äffchen mit dem Namen „To-Dewey“
- To-Dewey wird mit dem Benutzer interagieren
- To-Dewey wird über Push-Nachrichten verschiedene Updates geben
- Bei längeren Nicht-Nutzung der App wird To-Dewey den Benutzer „freundlich“ auf das Erfüllen derer Aufgaben Informieren
- Die Avatare der Benutzer werden ähnlich aussehen wie To-Dewey

Aufbauend auf diesen Entscheidungen wurde das Maskottchen & Logo erstellt und Hausfarben & Hausschriften gewählt.

### **Farben**
Bei den Farben haben wir uns für schlichte und sauber aussehende (helle und dunkle) Grautöne entschieden die von einem Türkis als Akzentfarbe unterstützt werden.
- Dark-Gray: #222831
- Mid-Gray: #393E46
- Light-Gray: #EEEEEE
- Dewey-Teal: #00ADB5

![design_kombi](https://github.com/user-attachments/assets/e9d0c7dd-2866-4820-833d-7f3ab3264843)

### **Schrift**
Für den Fließtext bzw. generell Text haben wir uns für eine einfache serifenlose Linear-Antiqua, für sauberes Aussehen und einfache Lesbarkeit, entschieden. Es handelt sich dabei um „Montserrat“. 

![Frame 3](https://github.com/user-attachments/assets/ac51aeab-5307-467d-95b9-6124d2c5cb37)

Als Scmuckschrift haben wir eine Schriftart der Klasse der Antiqua-Varianten gewählt, um eine herausstechende Wort-Bild-Marke erstellen zu können. Dabei handelt sich es um die etwas futuristisch aussehende „Nico Moji“. Diese wird zunächst ausschließlich für den Schriftzug des Logos bzw. der Marke in der App verwendet.

![Frame 1](https://github.com/user-attachments/assets/2170b97c-7809-40c8-86b6-c36e42e0ae6b)

## **Abschluss Sprint 3**
Auch diese Woche haben wir ein kurzes Retro zum Sprint gehalten. Im Großen und Ganzen sind wir sehr zufrieden mit unseren ersten richtige Fortschritten bei der Entwicklung der App. Die Arbeit am Projekt macht besonders unseren Entwicklern jetzt mehr Spaß, als in vorherigen Sprints. Wir haben uns auch schon einige Gedanken gemacht, wie wir diese Motivation aufrecht erhalten können und freuen uns schon auf die Weiterentwicklung von TDC im nächsten Sprint.

**Comment by MB-stack06** [comment]
Bei der ersten Betrachtung des Qualitätsbaums könnte man denken, dass er etwas knapp ausgeführt ist. Doch nach Einsicht der hinterlegten Links und der detaillierteren Ausführung wird deutlich, dass ihr die verschiedenen möglichen Szenarien gut einschätzt und ein fundiertes Verständnis für potenzielle Störfaktoren besitzt.

Noch eine kleine persönliche Randnotiz bezüglich der "a) / b)"-Thematik der hinterlegten Links zum Qualitätsbaum:

S01: a) => dünn                  b)=> dünn
S02: a) => fett                     b)=> dünn
S03: a) => fett                     b)=> dünn
S04: a)=> nicht vorhanden b)=> dünn
S05: a)=> nicht vorhanden b)=> dünn

Ist jedoch, wie gesagt, nur ein persönliches Anliegen und schmälert keineswegs eure dargebrachte Leistung.

Zudem müssen wir ebenfalls eure ADRs loben, die Entscheidungen eurerseits wurden mit Bedacht gewählt und sind anhand der Vor- und Nachteile perfekt nachzuvollziehen, womit sie kein Raum für Fragen lassen.
Auch wenn bei "RESTful API" der "asynchronen API-Aufrufen"-Link erneut hinterlegt ist, haben wir euer ADR 3 ebenfalls gefunden.

Als Schlusswort wollen wir euch zu euren exzellenten Fortschritten im Bereich Design gratulieren, uns gefällt besonders gut die Schriftart für euer Logo, welche einen sehr modernen Eindruck hinterlässt, was sich auch in eurer atemberaubenden Farbpalette widerspiegelt.

Wir freuen uns auf neue Updates und verfolgen euren Fortschritt begeistert.
Mit besten Grüßen, euer GymTrackerX-Team.

**Reply by Ninetilt** [reply]
Danke für das Feedback :)
Ich habe den Link zum ADR der RESTful API entsprechend angepasst und die Qualitätszenarien nochmal einheitlich formatiert.

Viele Grüße
Prokekt TDC (JS)

**Comment by TimHitschfeld** [comment]
Der Blogpost gibt einen spannenden Einblick in die Fortschritte und Entscheidungen, die im Rahmen des Projekts getroffen wurden. Besonders beeindruckend finde ich, wie strukturiert und durchdacht die Architektur- und Designansätze präsentiert werden.

Die Wahl der serviceorientierten Architektur in Kombination mit RESTful APIs und asynchronen Aufrufen zeigt, dass ihr euch auf Skalierbarkeit und Performance fokussiert. Auch die Qualitätsszenarien machen deutlich, dass ihr Nutzerbedürfnisse ernst nehmt und eine stabile, benutzerfreundliche und sichere App schaffen wollt.

Das Maskottchen „To-Dewey“ ist eine großartige Idee! Es verleiht der Marke Charakter und stärkt die Identität der App. Die Idee, durch Push-Benachrichtigungen und freundliche Erinnerungen mit den Nutzern zu interagieren, macht die App sicherlich lebendiger und einzigartig.

Die Designentscheidungen wirken ebenfalls sehr professionell. Die Farbpalette und die gewählten Schriftarten strahlen Modernität und Klarheit aus.

Viele Grüße
Euer TripAhead Team /TH
